torch/_functorch/_aot_autograd/utils.py

```diff
@@ -459,6 +449,26 @@ def _is_backward_node_with_seq_nr(node):
             node.meta["custom"] = fwd_node.meta.get("custom")
 
 
+def copy_fwd_metadata_to_bw_nodes(fx_g):
+    """
+    Input: `fx_g` which contains the joint fwd+bwd FX graph created by
+    aot_autograd.
+
+    This function walks the graph and copies over metadata from forward nodes
+    to backward nodes, using the `seq_nr` field as a one-to-many mapping
+    from forward node to backward node. This metadata is useful for performance
+    profiling and debugging.
+    """
+
+    # Copy the metadata recursively - useful for HOPs
+    for node in fx_g.graph.nodes:
+        if node.op == "get_attr":
+            submod = getattr(fx_g, node.target)
+            if isinstance(submod, torch.fx.GraphModule):
+                copy_fwd_metadata_to_bw_nodes(submod)
```

Comment:
do we know for sure that the backward nodes and the forward nodes are always in the same subgraph? If so this would work. In the current implementation, the bw nodes would not be able to find the fw nodes in other subgraphs.